rules/expiring-todo-comments.js

```diff
@@ -0,0 +1,357 @@
+'use strict';
+const readPkg = require('read-pkg');
+const semver = require('semver');
+const ci = require('ci-info');
+const baseRule = require('eslint/lib/rules/no-warning-comments');
+const getDocsUrl = require('./utils/get-docs-url');
+
+const pkg = readPkg.sync();
+
+const pkgDependencies = {...pkg.dependencies, ...pkg.devDependencies};
+
+const DEPENDENCY_INCLUSION_RE = /^[+|-]\s*@?[\S+]\/?\S+/;
+const DEPENDENCY_VERSION_RE = /^(@?[\S+]\/?\S+)@(>|>=)([\d]+(\.\d+){0,2})/;
+const PKG_VERSION_RE = /^(>|>=)([\d]+(\.\d+){0,2})\s*$/;
+const ENGINES_RE = /^engines (\S+)(>|>=)([\d]+(\.\d+){0,2})/;
+const ISO8601 = /(\d{4})-(\d{2})-(\d{2})/;
+
+const create = context => {
+	const options = {
+		terms: ['todo', 'fixme', 'xxx'],
+		ignoreDatesOnPR: true,
+		...context.options[0]
+	};
+
+	const sourceCode = context.getSourceCode();
+	const comments = sourceCode.getAllComments();
+	const unnusedComments = comments
+		.filter(token => token.type !== 'Shebang')
+		.filter(processComment);
+
+	// This is highly dependable on eslint's no-warning-comments implementation.
+	// What I do is patch the parts I know the rule will use, the getAllComments.
+	// Since I have priority, I leave only the comments that I didn't use to it.
+	const fakeContext = {...context, getSourceCode() {
+		return {...sourceCode, getAllComments() {
+			return unnusedComments;
+		}};
+	}};
+	const rules = baseRule.create(fakeContext);
+
+	function processComment(comment) {
+		const parsed = parseTodoWithArgs(comment.value, options);
+
+		if (!parsed) {
+			return true;
+		}
+
+		// Count if there where valid props
+		// Otherwise it's a useless TODO (fallback to no-warning-comments)
+		let uses = 0;
+
+		const {
+			packageVersions = [],
+			dates = [],
+			dependencies = [],
+			engines = []
+		} = parsed;
+
+		if (dates.length > 1) {
+			uses++;
+			context.report({
+				node: null,
+				loc: comment.loc,
+				messageId: 'avoidMultipleDates',
```

Comment:
Can you move the message ids to constants at the top of the file? E.g., `const MESSAGE_ID_MULTIPLE_DATES = 'avoidMultipleDates';`

Reply:
I've missed this, thanks for pointing out. Now fixed.